compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1310,11 +1310,25 @@ class Definitions {
 
   private lazy val ScalaNumericValueTypes: collection.Set[TypeRef] = ScalaNumericValueTypeList.toSet
   private lazy val ScalaValueTypes: collection.Set[TypeRef] = ScalaNumericValueTypes + UnitType + BooleanType
-  private lazy val ScalaBoxedTypes = ScalaValueTypes map (t => boxedTypes(t.name))
 
   val ScalaNumericValueClasses: PerRun[collection.Set[Symbol]] = new PerRun(implicit ctx => ScalaNumericValueTypes.map(_.symbol))
   val ScalaValueClasses: PerRun[collection.Set[Symbol]]        = new PerRun(implicit ctx => ScalaValueTypes.map(_.symbol))
-  val ScalaBoxedClasses: PerRun[collection.Set[Symbol]]        = new PerRun(implicit ctx => ScalaBoxedTypes.map(_.symbol))
+
+  private lazy val ScalaBoxedTypes: SimpleIdentityMap[Symbol, TypeRef] =
```

Comment:
Shouldn't the key of the map be context dependant? Make it a `PerRun`?

Reply:
Actually it is only used in `def boxedType`. Can you inline it?
```scala
def boxedType(tp: Type)(implicit ctx: Context): TypeRef = tp match {
  case ByteType    => BoxedByteType
  case ShortType   => BoxedShortType
  case CharType    => BoxedCharType
  case IntType     => BoxedIntType
  case LongType    => BoxedLongType
  case FloatType   => BoxedFloatType
  case DoubleType  => BoxedDoubleType
  case UnitType    => BoxedUnitType
  case BooleanType => BoxedBooleanType
}
```

Reply:
matching on the type is dangerous since there might be multiple instances of `Type` which are `=:=` but not `==`, but doing a match on the classSymbol would work and is simpler indeed so I'll switch to that.